### PR TITLE
feat: setup compatibility test will make sure to catch when prealignment and alignment aren't compatible.

### PR DESCRIPTION
**Pull request by Smeds**
### Review and tests: 
- [ ] Tests pass

### CODE REVIEW

**Review by jonca79 — APPROVED** [review]

File: .tests/compatibility/resources.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+default_resources:
+  threads: 1
+  time: "12:00:00"
+  mem_mb: 1024
+  mem_per_cpu: 3072
+  partition: core
+
+bwa_mem:
```
Comment (marrip) [inline .tests/compatibility/resources.yaml:8]: Do we need this?

File: .tests/compatibility/Snakefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        "alignment/samtools_merge_bam/NA12878_N.bam",
+
+
+configfile: "config.yaml"
+
+
+config = load_resources(config, config["resources"])
+
+
+def extract_module_version_from_readme(modulename):
```
Comment (marrip) [inline .tests/compatibility/Snakefile:17]: Smart 😉 🎉

File: .github/workflows/compatibility.master.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Compatibility test master
```
Comment (marrip) [inline .github/workflows/compatibility.master.yaml:1]: Do we still need both these files?

Comment (Smeds) [reply .github/workflows/compatibility.master.yaml]: nope! I will remove it

File: .tests/compatibility/Snakefile (hunk 2 above)
Comment (Smeds) [reply .tests/compatibility/Snakefile:17]: en enkel lösning till att börja med! Så nu behöver bara README uppdateras=)

File: .tests/compatibility/resources.yaml (hunk 1 above)
Comment (Smeds) [reply .tests/compatibility/resources.yaml]: I have  now removed it!

**Comment by Smeds** [comment]
@marrip  I updated the pull-request with a COMPATIBILITy.md file

File: .tests/compatibility/resources.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+default_resources:
+  threads: 1
+  time: "12:00:00"
+  mem_mb: 1024
+  mem_per_cpu: 3072
```
Comment (marrip) [inline .tests/compatibility/resources.yaml:5]: mem_per_cpu should be the same as mem_mb

File: COMPATIBILITY.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# Compatibility 
+## hydra-genetics/prealignment
+
+| prealignment version | alignment version |
+| --- | --- |
+| v0.2.0 |  |
```
Comment (marrip) [inline COMPATIBILITY.md:6]: This one will be updated as soon as we have a release right? But how do we do that? It needs to be set the second we release kind of.

Comment (Smeds) [reply COMPATIBILITY.md:6]: We will have a procedure where one updates it manual before doing the release, at least until we find a automatic way of doing it

File: .tests/compatibility/resources.yaml (hunk 4 above)
Comment (Smeds) [reply .tests/compatibility/resources.yaml]: fixed

**Review by marrip — APPROVED** [review]

File: COMPATIBILITY.md (hunk 5 above)
Comment (marrip) [reply COMPATIBILITY.md:6]: ok 👍

**Pull request merged**